src/server/game/Movement/MovementGenerators/WaypointMovementGenerator.cpp

```diff
@@ -313,7 +349,12 @@ void FlightPathMovementGenerator::DoFinalize(Player* player)
     // remove flag to prevent send object build movement packets for flight state and crash (movement generator already not at top of stack)
     player->ClearUnitState(UNIT_STATE_IN_FLIGHT);
 
-    // xinef: this should be cleaned by CleanupAfterTaxiFlight(); function!
+    uint32 taxiNodeId = player->m_taxi.GetTaxiDestination();
+    if (!player->IsInWorld())
```

Comment:
if (!player->IsInWorld()) Is this wrong?

Reply:
After the flight is completed and lands, the error I mentioned above is caused by the error here